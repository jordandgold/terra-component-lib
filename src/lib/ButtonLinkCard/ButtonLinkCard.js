import React from "react";
import "./ButtonLinkCard.scss";
import PropTypes from "prop-types";
import ButtonLink from "../ButtonLink/ButtonLink";

const ButtonLinkCard = ({ content }) => {
  return (
    <article className="ter-button-link-card">
      <h4 className="ter-button-link-card--title">{content.title}</h4>
      <p className="ter-button-link-card--body">{content.body}</p>
      <ButtonLink
        link={content.link.url}
        text={content.link.text}
        className="ter-button--primary--1"
      />
    </article>
  );
};

export default ButtonLinkCard;

ButtonLinkCard.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    link: PropTypes.shape({
      url: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  })
};
