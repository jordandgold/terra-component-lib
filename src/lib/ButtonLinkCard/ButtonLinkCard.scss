@import "../global-styles/terra.scss";

.ter-button-link-card {
  max-width: 500px;
  padding: 24px;

  &--title {
    font-size: font-size("md");
  }
}
