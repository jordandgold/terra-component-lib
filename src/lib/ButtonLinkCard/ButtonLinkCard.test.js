import React from "react";
import ButtonLinkCard from "./ButtonLinkCard";
import { shallow } from "enzyme";

describe("ButtonLinkCard", () => {
  let wrapper;

  beforeEach(() => {
    const mockContent = {
      title: "Design is good. Period.",
      body:
        "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!",
      link: {
        url: "#",
        text: "Button"
      }
    };
    wrapper = shallow(<ButtonLinkCard content={mockContent} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
