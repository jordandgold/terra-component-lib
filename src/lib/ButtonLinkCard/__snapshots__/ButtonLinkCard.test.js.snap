// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLinkCard should match the snapshot 1`] = `
<article
  className="ter-button-link-card"
>
  <h4
    className="ter-button-link-card--title"
  >
    Design is good. Period.
  </h4>
  <p
    className="ter-button-link-card--body"
  >
    Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!
  </p>
  <ButtonLink
    className="ter-button--primary--1"
    link="#"
    text="Button"
  />
</article>
`;
